src/tabbar.cpp

```diff
@@ -174,6 +174,11 @@ void TabBar::openUrl(const QUrl& url, bool newTab)
     }
     QUITIFNULL(webView);
     webView->setUrl(url);
+#if QT_VERSION < QT_VERSION_CHECK(5, 13, 0) // Earlier than Qt 5.13
+    webView->page()->profile()->setRequestInterceptor(ExternalReqInterceptor::instance());
+#else // Qt 5.13 and later
+    webView->page()->setUrlRequestInterceptor(ExternalReqInterceptor::instance());
```

Comment:
Why don't you install the interceptor on the profile (rather than page) in Qt 5.13 and newer?

Note that if you install the interceptor on the profile, you can do it only once (e.g. in `KiwixApp::init()`) rather than for each new page.

Reply:
Was not aware of our Kprofile class. Then, I believe this class should be rather implemented in src/kprofile.h.

Reply:
I was not aware of `KProfile` either.

> I am inclined to just define this class in src/kprofile.cpp since nobody else uses it anyways and we can now abandon the singleton.

Sounds good

Reply:
Nvm, you cannot declare QObjects in .cpp files so I will just move the definition to .h.